components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1187,9 +1187,16 @@ void RewardsServiceImpl::LoadNicewareList(
 void RewardsServiceImpl::SavePublisherInfo(
     ledger::PublisherInfoPtr publisher_info,
     ledger::PublisherInfoCallback callback) {
+  DCHECK(publisher_info);
+  if (!publisher_info) {
+    callback(ledger::Result::LEDGER_ERROR, std::move(publisher_info));
+    return;
+  }
+
+  auto copy = publisher_info->Clone();
   base::PostTaskAndReplyWithResult(file_task_runner_.get(), FROM_HERE,
       base::BindOnce(&SavePublisherInfoOnFileTaskRunner,
-                    publisher_info->Clone(),
+                    std::move(copy),
```

Comment:
If a developer is copy/pasting there is potential for a bug, I would personnaly clone inline for each `std::bind`